Source: Ali Polatel <alip@exherbo.org>
Upstream: No (urkud promised to take it upstream)
Reason: Fail in configure test for abstract sockets

--- dbus-1.2.16/configure.in	2010-02-25 20:19:44.000000000 +0200
+++ dbus-1.2.16/configure.in	2010-02-25 20:21:28.000000000 +0200
@@ -590,6 +590,7 @@
 #include <errno.h>
 ]],
 [[
+  size_t slen;
   int listen_fd;
   struct sockaddr_un addr;
   
@@ -604,9 +605,13 @@
   memset (&addr, '\0', sizeof (addr));
   addr.sun_family = AF_UNIX;
   strcpy (addr.sun_path, "X/tmp/dbus-fake-socket-path-used-in-configure-test");
+  /* SUN_LEN uses strlen() so need to calculate it before adding \0 at the
+   * beginning.
+   */
+  slen = SUN_LEN(&addr);
   addr.sun_path[0] = '\0'; /* this is what makes it abstract */
   
-  if (bind (listen_fd, (struct sockaddr*) &addr, SUN_LEN (&addr)) < 0)
+  if (bind (listen_fd, (struct sockaddr*) &addr, slen) < 0)
     {
        fprintf (stderr, "Abstract socket namespace bind() failed: %s\n", 
                 strerror (errno));
